src/derivable/map.spec.ts

```diff
@@ -0,0 +1,167 @@
+            const stack = d$['_stack'];
+            expect(() => d$.get()).to.throw('the Error');
+            expect(console.error).to.have.been.calledOnce
+                .and.to.have.been.calledWithExactly('the Error', stack);
+        });
+    });
+
+    it('should not call the deriver when the cached value is known to be up to date because of a reactor', () => {
+        const deriver = spy(() => 123);
+        const d$ = constant(0).map(deriver);
+        d$.get();
+        expect(deriver).to.have.been.calledOnce;
+        d$.react(() => 0);
+        expect(deriver).to.have.been.calledTwice;
+        d$.get();
+        expect(deriver).to.have.been.calledTwice;
+    });
+
+    it('should disconnect when no longer used', () => {
+        const a$ = atom(1);
+        const m$ = a$.map(v => v);
+        const d$ = m$.derive(v => v);
+
+        expect(d$.connected).to.be.false;
+        expect(m$.connected).to.be.false;
+        expect(a$.connected).to.be.false;
+
+        const stopReaction = d$.react(() => 0);
+
+        expect(d$.connected).to.be.true;
+        expect(m$.connected).to.be.true;
+        expect(a$.connected).to.be.true;
+
+        stopReaction();
+
+        expect(d$.connected).to.be.false;
+        expect(m$.connected).to.be.false;
+        expect(a$.connected).to.be.false;
+    });
+
+    it('should call the deriver again when the cached value is known not to be up to date', () => {
+        const deriver = spy((n: number) => n * 2);
+        const a$ = atom(1);
+        const m$ = a$.map(deriver);
+        const d$ = m$.derive(v => v);
+        const stopReaction = d$.react(() => 0);
+        expect(deriver).to.have.been.calledOnce;
+        expect(d$.get()).to.equal(2);
+        expect(deriver).to.have.been.calledOnce;
+        stopReaction();
+        expect(deriver).to.have.been.calledOnce;
+        a$.set(2);
+        d$.get();
+        expect(deriver).to.have.been.calledTwice;
+    });
+
+    it('should cache thrown errors to rethrow them on multiple accesses until the derivation produces a new result', () => {
+        const a$ = atom(false);
+        const theError = new Error('the error');
+        const deriver = spy((a: boolean) => { if (a) { throw theError; } else { return 'a value'; } });
+        const d$ = a$.map(deriver).autoCache();
+        expect(d$.get(), 'first time').to.equal('a value');
+        expect(d$.get(), 'second time').to.equal('a value');
+        expect(deriver).to.have.been.calledOnce;
+        a$.set(true);
+        expect(() => d$.get(), 'first time').to.throw(theError);
+        expect(() => d$.get(), 'second time').to.throw(theError);
+        expect(deriver).to.have.been.calledTwice;
+        a$.set(false);
+        expect(d$.get(), 'first time').to.equal('a value');
+        expect(d$.get(), 'second time').to.equal('a value');
+        expect(deriver).to.have.been.calledThrice;
+    });
+
+    it('should allow error objects as valid values', () => {
```

Comment:
Should this be a standard test in `testDerivable`?

Reply:
I think we can just remove this test. This was before we added the `State<V>` type.